src/Microsoft.Azure.WebJobs.Logging/FunctionId.cs

```diff
@@ -0,0 +1,108 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System;
+
+namespace Microsoft.Azure.WebJobs.Logging
+{
+    /// <summary>
+    /// Strongly typing for a function Id. 
+    /// FunctionIds include HostName and FunctionName
+    /// </summary>
+    public struct FunctionId
+    {
+        internal string Value { get; set; }
+
+        /// <summary>
+        /// Get string representation. This will be consist of valid Azure Table row key characters. 
+        /// </summary>
+        /// <returns></returns>
+        public override string ToString()
+        {
+            return this.Value;
+        }
+
+        /// <summary>
+        /// Throw exception if this instance is bad. 
+        /// </summary>
+        public void Validate()
+        {
+            if (string.IsNullOrEmpty(this.Value))
+            {
+                throw new InvalidOperationException("FunctionId is not valid.");
+            }
+        }
+
+        /// <summary>
+        /// Build directly from a string. This skips escaping and should only be used by the reader for interop. 
+        /// This is the direct inverse of ToString().
+        /// </summary>
+        /// <param name="value">raw value from a previos ToString() call. </param>
+        /// <returns></returns>
+        public static FunctionId DangerousFromString(string value)
```

Comment:
Can you remove this "Dangerous" terminology from the public API - it's strange? E.g., since this is the inverse of ToString, How about a Parse or TryParse method, which is a common BCL pattern. Or perhaps just a Build overload that takes a string?